packages/mobile/src/components/challenge-rewards-drawer/ChallengeRewardsDrawerProvider.tsx

```diff
@@ -104,16 +104,18 @@ export const ChallengeRewardsDrawerProvider = () => {
   const AAOEndpoint = useRemoteVar(StringKeys.ORACLE_ENDPOINT)
   const hasConfig = (oracleEthAddress && AAOEndpoint && quorumSize > 0) || true
   const onClaim = useCallback(() => {
-    dispatch(
-      claimChallengeReward({
-        claim: {
-          challengeId: modalType,
-          specifiers: [challenge?.specifier ?? ''],
-          amount: challenge?.amount ?? 0
-        },
-        retryOnFailure: true
-      })
-    )
+    if (challenge) {
+      dispatch(
+        claimChallengeReward({
+          claim: {
+            challengeId: modalType,
+            specifiers: { [challenge.specifier]: challenge.amount },
+            amount: challenge?.claimableAmount ?? 0
+          },
+          retryOnFailure: true
+        })
+      )
+    }
```

Comment:
what if not challenge?

Reply:
then don't do anything.. but if not challenge then the tile for that challenge wouldn't have existed in the first place.